core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -325,8 +325,21 @@ public void run() {
     private <ReqT, RespT> ServerStreamListener startCall(ServerStream stream, String fullMethodName,
         ServerMethodDefinition<ReqT, RespT> methodDef, Future<?> timeout,
         Metadata headers) {
+
+      if (headers.containsKey(GrpcUtil.MESSAGE_ENCODING_KEY)) {
+        String messageEncoding = headers.get(GrpcUtil.MESSAGE_ENCODING_KEY);
```

Comment:
Should you just get it rather than call `containsKey` to avoid the second lookup?